src/pages/SearchResults.js

```diff
@@ -0,0 +1,223 @@
+            // Include genreQuery for the 'genre' filter as well
+            let genreQuery = '';
+            if (genres && genres.length > 0) {
+                genreQuery = genres.join('+');
+            }
+            console.log('genreQuery:', genreQuery);
+            apiUrl = `https://www.googleapis.com/books/v1/volumes?q=${encodedQuery}+subject:${genreQuery}&orderBy=newest&maxResults=3`;
+        }
+    
+        if (apiUrl) {
+            console.log('API URL:', apiUrl);
+            axios.get(apiUrl)
+                .then(res => {
+                    console.log('API response:', res); // Log the entire response object
+                    if (res.data && res.data.items) {
+                        const items = res.data.items;
+                        let resultList = [];
+                        if (filter === 'all' || filter === 'titles') {
+                            resultList = items.map(item => ({
+                                id: item.id,
+                                title: item.volumeInfo.title,
+                                authors: item.volumeInfo.authors ? item.volumeInfo.authors.join(', ') : 'Unknown Author',
+                            }));
+                        } else if (filter === 'authors') {
+                            resultList = items.map(item => ({
+                                id: item.id,
+                                name: item.volumeInfo.authors ? item.volumeInfo.authors.join(', ') : 'Unknown Author',
+                            }));
+                        } else if (filter === 'genre') {
+                            resultList = items.map(item => ({
+                                id: item.id,
+                                title: item.volumeInfo.title,
+                                authors: item.volumeInfo.authors ? item.volumeInfo.authors.join(', ') : 'Unknown Author',
+                            }));
+                        }
+                        setSearchResults(resultList);
+                    }
+                })
+                .catch(err => console.error('Error fetching data:', err));
+        } else {
+            // handle other search filters
+            if (filter === 'users') {
+                // Handle user search by searching usernames in firebase
+                console.log('Searching for users...' + query);
+                const uid = await fetchUIDwithUsername(query.toLowerCase());
+                console.log('UID:', uid);
+                searchUsers(uid);
+            }
+        }
+    };
+    
+    
+    
+    
+    
+    
+    
+    const handleFilterChange = event => {
+        setSelectedFilter(event.target.value);
+    };
+
+    const handleSubmit = event => {
+        event.preventDefault();
+        performSearch(searchQuery, selectedFilter, selectedGenres); // Include selectedGenre in the call to performSearch
+    };
+
+    return (
+        <div className="flex">
+            {/* Display search results */}
```

Comment:
You can use the card element that I created in Sprint 1 to display results. If that's not a priority for this pull request, that's okay.

Reply:
I was definitely planning on it! Just spent a while to get the filtering and getting used to the api calls